src/components/players/player/player.tsx

```diff
@@ -1,27 +1,100 @@
-import { type FC } from "react";
-import { Avatar, Stack, Typography } from "@mui/material";
+import React, { type FC, useEffect, useState } from "react";
+import { Stack, Typography } from "@mui/material";
 
 import { styles } from "./styles.js";
 import { Skeleton } from "./skeleton.jsx";
+import { type Character, type MatchStage } from "~/types/index.js";
+import { text } from "~/assets/text/index.js";
+import { CharacterAvatar } from "~/components/character-avatar/index.js";
 
 interface Props {
-  color?: string;
-  characterName: string;
+  character: Character;
+  selected?: boolean;
+  isLocalPlayer?: boolean;
+  hasGuessed?: boolean;
+  isBot?: boolean;
+  onSelectPlayer?: () => void;
+  stage?: MatchStage;
 }
 
-export const Player: FC<Props> = ({ color = "blue", characterName }) => {
-  // TODO: Fix "src" after svg's set up
-  const src = "../images/svg/alien.svg";
+export const Player: FC<Props> = ({
+  character,
+  isLocalPlayer = false,
+  selected,
+  hasGuessed,
+  isBot,
+  onSelectPlayer,
+  stage = "chat",
+}) => {
+  const { name, color } = character;
   const textColor = `${color}.dark`;
+  const [isBotSelect, setIsBotSelect] = useState<boolean | undefined>(false);
 
-  if (!characterName) return <Skeleton />;
+  useEffect(() => {
+    if (stage === "voting" && selected) {
+      setIsBotSelect(true);
+    } else {
+      setIsBotSelect(false);
+    }
+  }, [selected, stage]);
+
+  if (!name) return <Skeleton />;
+  let textResult = isBot ? text.match.isBot : text.match.isHuman;
+
+  if (stage === "results") {
+    if (selected === true && isBot === true) {
+      textResult = text.match.botBusted;
+    } else if (selected === true && isBot === false) {
+      textResult = text.match.isNotBot;
+    } else if (selected === false && isBot === false) {
+      textResult = text.match.isHuman;
+    }
+  }
 
   return (
     <Stack sx={styles.container}>
-      <Avatar src={src} color={color} sx={styles.avatar} />
-      <Typography variant="body1" color={textColor}>
-        {characterName}
-      </Typography>
+      {stage !== "chat" && (
+        <Typography variant="body1" sx={styles.character} color={textColor}>
+          {name}
+        </Typography>
+      )}
+      <CharacterAvatar
+        stage={stage}
+        character={character}
+        isSelected={selected}
+        hasGuessed={hasGuessed}
+        isBot={isBot}
+        onSelectPlayer={onSelectPlayer}
+      />
+      {stage === "chat" && (
+        <>
+          <Typography variant="body1" sx={styles.character} color={textColor}>
+            {name}
+          </Typography>
+          {isLocalPlayer && (
+            <Typography variant="body1" sx={styles.character} color={textColor}>
+              {text.match.localPlayer}
+            </Typography>
+          )}
+        </>
+      )}
+      {stage === "voting" && isBotSelect && (
```

Comment:
isBotSelect is already checking for `stage === "voting"`.
Can you make the logic DRYer? It should help for general readability of whats going on.